### ISSUE TITLE
Cluster-autoscaler "Ignoring in scale up" despite aws-ebs-csi-driver volume-attach-limit is reached

### ISSUE THREAD

**Issue by tulsluper** [kubernetes/kubernetes#106894]
### What happened?

We use [aws-ebs-csi-driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver) on nodes with the arg `--volume-attach-limit=24`

> volume-attach-limit - Value for the maximum number of volumes attachable per node. If specified, the limit applies to all nodes. If not specified, the value is approximated from the instance type.

[AWS Nitro System volume limits](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html): 

> Instances built on the Nitro System support a maximum number of attachments, which are shared between network interfaces, EBS volumes, and NVMe instance store volumes. ... Most of these instances support a maximum of 28 attachments. For example, if you have no additional network interface attachments on an EBS-only instance, you can attach up to 27 EBS volumes to it. If you have one additional network interface on an instance with 2 NVMe instance store volumes, you can attach 24 EBS volumes to it.

It's why we set `--volume-attach-limit=24` (without it, if volumesInUse for a node is already 24, a new volume is assigned to the node but cannot be attached)

The issue is that cluster-autoscaler ignores this limit when evaluating which nodes are suitable for scheduling pods.

E.g.
```
NAME                            READY   STATUS    RESTARTS   AGE
vl-0gcm7ndm-redis-ha-server-0   0/2     Pending   0          120m
vl-5pkkpgnz-redis-ha-server-0   0/3     Pending   0          120m
vl-kblxoduk-redis-ha-server-1   0/3     Pending   0          120m
vl-nmqtjs17-redis-ha-server-0   0/2     Pending   0          120m
```
During 2 hours (until the ASG was scaled manually) there were autoscaler messages like:
`
Pod here-olp-volatilelayer-olp-cn-here-prod.vl-kblxoduk-redis-ha-server-1 marked as unschedulable can be scheduled on node ip-10-252-87-54.cn-northwest-1.compute.internal (based on hinting). Ignoring in scale up
` (**Ignoring in scale up**)

This node (ip-10-252-87-54) had maximum number of volumesInUse - 24
```
❯ kubectl get nodes ip-10-252-87-54.cn-northwest-1.compute.internal -o yaml | grep "\- kubernetes.io/csi/ebs.csi.aws.com" | wc -l
24
```

We face this issue from time to time.
Can it be fixed?

Kubernetes: v1.20.13
aws-ebs-csi-driver:v1.2.0
cluster-autoscaler:v1.20.1

### What did you expect to happen?

a pod can't be scheduled on a node if volume-attach-limit is reached;
don't block scaling up;

### How can we reproduce it (as minimally and precisely as possible)?

- node - AWS nitro instance type; in node group managed by cluster-autoscaler
- aws-ebs-csi-driver with `--volume-attach-limit=xx`
- create n number of pods/PVCs with nodeSelector of this node to reach volumesInUse == volume-attach-limit
- create one more the same pod/PVCs -> check autoscaler logs for this pod and "Ignoring in scale up"

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T21:15:16Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.13", GitCommit:"2444b3347a2c45eb965b182fb836e1f51dc61b70", GitTreeState:"clean", BuildDate:"2021-11-17T13:00:29Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
AWS
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>
aws-ebs-csi-driver:v1.2.0

cluster-autoscaler:v1.20.1
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by ialidzhikov** [comment]
Seems to be the same issue I reporter - https://github.com/gardener/gardener/issues/5064. I had plans to report this issue to the k/autoscaler repo in the next days.

**Comment by ialidzhikov** [comment]
> I had plans to report this issue to the k/autoscaler repo in the next days.

Ref https://github.com/kubernetes/autoscaler/issues/4517

**Comment by ialidzhikov** [comment]
I hope that this issue should be fixed with `cluster-autoscaler@v1.23` when you specify CSI migration related feature flags to the cluster-autoscaler. For more details you can check https://github.com/kubernetes/autoscaler/pull/4539.
